hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -0,0 +1,897 @@
+                "domain": {
+                    "add_line_task_id": [
+                        ("project_id", "=", self.add_line_project_id.id),
+                        ("company_id", "=", self.company_id.id),
+                        ("id", "not in", self.timesheet_ids.mapped("task_id").ids),
+                    ]
+                }
+            }
+        else:
+            return {"domain": {"add_line_task_id": [("id", "=", False)]}}
+
+    @api.model
+    def _check_employee_user_link(self, vals):
+        if "employee_id" in vals:
+            employee = self.env["hr.employee"].sudo().browse(vals["employee_id"])
+            if not employee.user_id:
+                raise UserError(
+                    _(
+                        "In order to create a sheet for this employee, you must"
+                        " link him/her to an user: %s"
+                    )
+                    % (employee.name,)
+                )
+            return employee.user_id.id
+        return False
+
+    def copy(self, default=None):
+        if not self.env.context.get("allow_copy_timesheet"):
+            raise UserError(_("You cannot duplicate a sheet."))
+        return super().copy(default=default)
+
+    @api.model
+    def create(self, vals):
+        self._check_employee_user_link(vals)
+        res = super().create(vals)
+        res.write({"state": "draft"})
+        return res
+
+    def _sheet_write(self, field, recs):
+        self.with_context(sheet_write=True).write({field: [(6, 0, recs.ids)]})
+
+    def write(self, vals):
+        self._check_employee_user_link(vals)
+        res = super().write(vals)
+        for rec in self:
+            if rec.state == "draft" and not self.env.context.get("sheet_write"):
+                rec._update_analytic_lines_from_new_lines(vals)
+                if "add_line_project_id" not in vals:
+                    rec.delete_empty_lines(True)
+        return res
+
+    def unlink(self):
+        for sheet in self:
+            if sheet.state in ("confirm", "done"):
+                raise UserError(
+                    _(
+                        "You cannot delete a timesheet sheet which is already"
+                        " submitted or confirmed: %s"
+                    )
+                    % (sheet.complete_name,)
+                )
+        return super().unlink()
+
+    def _get_informables(self):
+        """ Hook for extensions """
+        self.ensure_one()
+        return self.employee_id.parent_id.user_id.partner_id
+
+    def _get_subscribers(self):
+        """ Hook for extensions """
+        self.ensure_one()
+        subscribers = self._get_possible_reviewers().mapped("partner_id")
+        subscribers |= self._get_informables()
+        return subscribers
+
+    def _timesheet_subscribe_users(self):
+        for sheet in self.sudo():
+            subscribers = sheet._get_subscribers()
+            if subscribers:
+                self.message_subscribe(partner_ids=subscribers.ids)
```

Comment:
self.message_subscribe(partner_ids=subscribers.ids) should be sheet.message_subscribe(partner_ids=subscribers.ids)